### PR TITLE
OAPIF:  INITIAL_REQUEST_PAGE_SIZE  (fixes #4556) 

### LINKED ISSUE #4556: OAPIF: add a INITIAL_REQUEST_PAGE_SIZE open option
to override the general PAGE_SIZE open option which override the server default setting.
The intent is to be able to have a quick way of establishing the layer schema.
See https://lists.osgeo.org/pipermail/gdal-dev/2021-September/054711.html for related thread

### PR DESCRIPTION
**Pull request by elpaso**
Add an open option for OAPIF to set INITIAL_REQUEST_PAGE_SIZE, default is also set to 20

Fixes #4556

### CODE REVIEW

File: ogr/ogrsf_frmts/wfs/ogroapifdriver.cpp [hunk 1]
```diff
@@ -1088,6 +1089,14 @@ bool OGROAPIFDataset::Open(GDALOpenInfo *poOpenInfo)
         m_bPageSizeSetFromOpenOptions = true;
     }
 
+    const int initialRequestPageSize = atoi(
+        CSLFetchNameValueDef(poOpenInfo->papszOpenOptions, "INITIAL_REQUEST_PAGE_SIZE", "-1"));
+
+    if ( initialRequestPageSize > 1 )
```
Comment (rouault) [inline ogr/ogrsf_frmts/wfs/ogroapifdriver.cpp:1095]:
```suggestion
    if ( initialRequestPageSize >= 1 )
```

**Comment by rouault** [comment]
formatting issues. You need to setup pre-commit

**Comment by rouault** [comment]
autotest/gdrivers/ogcapi.py needs to be updated

**Comment by elpaso** [comment]
> formatting issues. You need to setup pre-commit

sorry, I switched to a new laptop machine for today and forgot to set it up.

**Comment by rouault** [comment]
still formatting issues: run 'pre-commit run --all-files'

**Pull request merged**